packages/dynamic-links/lib/modular/index.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2016-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { firebase } from '..';
+
+export const ShortLinkType = {
+  SHORT: 'SHORT',
+  UNGUESSABLE: 'UNGUESSABLE',
+  DEFAULT: 'DEFAULT',
+};
+
+/**
+ * @returns {import('.').DynamicLinks}
+ */
+export function getDynamicLinks() {
+  return firebase.dynamicLinks();
+}
+
+/**
+ *
+ * @param {import('.').DynamicLinks} dynamicLinks
+ * @param {import('..').FirebaseDynamicLinksTypes.DynamicLinkParameters} dynamicLinkParams
+ * @returns {Promise<string>}
+ */
+export function buildLink(dynamicLinks, dynamicLinkParams) {
+  return dynamicLinks.buildLink(dynamicLinkParams);
+}
+
+/**
+ *
+ * @param {import('.').DynamicLinks} dynamicLinks
+ * @param {import('..').FirebaseDynamicLinksTypes.DynamicLinkParameters} dynamicLinkParams
+ * @param {import('..').FirebaseDynamicLinksTypes.ShortLinkType | undefined} shortLinkType
+ * @returns {Promise<string>}
+ */
+export function buildShortLink(dynamicLinks, dynamicLinkParams, shortLinkType) {
+  return dynamicLinks.buildShortLink(dynamicLinkParams, shortLinkType);
+}
+
+/**
+ * @param {import('.').DynamicLinks} dynamicLinks
+ * @returns {Promise<import('..').FirebaseDynamicLinksTypes.DynamicLink | null>}
+ */
+export function getInitialLink(dynamicLinks) {
+  return dynamicLinks.getInitialLink();
+}
+
+/**
+ * @param {import('.').DynamicLinks} dynamicLinks
+ * @param {(link: import('..').FirebaseDynamicLinksTypes.DynamicLink) => void} listener
+ * @returns {() => void}
+ */
+export function onLink(dynamicLinks, listener) {
+  return dynamicLinks.onLink(listener);
```

Comment:
This missing test annotation makes sense, I had to specifically disable onLink in CI because it's flaky there - sometimes the app isn't registered / allowed by the Android OS to be the app that handles the links, so it just hangs there waiting for a tap

✅